### ISSUE TITLE
FileNotFoundError: docker-compose.yml not found

### ISSUE THREAD

**Issue by vteivans** [EugenMayer/docker-sync#641]
### Error/Feature Requestion/Docs
https://github.com/EugenMayer/docker-sync/issues/582

### OS
Windows 10

### Docker Driver
WSL + Docker for Windows (18.09.1, build 4c52b90), docker-sync-0.5.7, docker-compose 2

### Sync strategy
Unison

### Your docker-sync.yml
```yml
version: "2"

options:
  verbose: false
  max-attempt: 30
syncs:
  #IMPORTANT: ensure this name is unique and does not match your other application container name
  app-server-sync: # tip: add -sync and you keep consistent names as a convention
    src: './server'
    sync_args: ['-perms=0'] #required for two way sync ie generators, etc
    sync_strategy: 'unison'
    # sync_user: 'root'
    sync_excludes: [
      '.idea/*',
      '.git/*',
      '.gitignore',
      'node_modules',
      '*/node_modules',
      '*/.gitignore',
      'docker-*',
      'Dockerfile',
      'Dockerfile-dev',
      '*/*.log',
      '*.log']
    sync_host_ip: '127.0.0.1' #host ip isn't properly inferred
```

### OS
Windows 10 1803

### Hey!
I'm not sure if this is the correct way to do this. I wanted to add a comment on the given issue, but it's locked. I ran in to it just today and once before. And there are two ways I have worked around this:

1. Install Linux Docker on WSL. For me at least, this problem only appears if I symlink the Windows docker binaries.

2. Simply running `docker-sync start` and `docker-compose -f docker-compose.yml -f docker-compose-dev.yml up` separately. No need to run one in bash and the other in PowerShell (at least for me). Simply run:
    - `docker-sync start`
    - `docker-compose -f docker-compose.yml -f docker-compose-dev.yml up`

### P.S.
Huge thanks to you guys for having this available on Windows. Most syncing stuff usually isn't. If I have some time I'll check the ruby code, maybe get some idea why is this happening.

**Comment by EugenMayer** [comment]
Could you please try upgrading to 0.5.11-beta2 and see if that helps?

**Comment by vteivans** [comment]
Thanks for the response @EugenMayer!
Tried the `0.5.11-beta2`, same thing, says the file is not there, even though it is. And running `docker-sync` and `docker-compose` separately still works correctly.

```log
.FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/user/Documents/proj/app/new-app/docker-compose.yml'
#<Thread:0x00007fffe77b4a00@/var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/tasks/stack/stack.thor:50 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        3: from /var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/tasks/stack/stack.thor:51:in `block in start'
        2: from /var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/lib/docker-sync/compose.rb:42:in `run'
        1: from /var/lib/gems/2.5.0/gems/docker-compose-1.1.10/lib/docker/compose/session.rb:94:in `up'
/var/lib/gems/2.5.0/gems/docker-compose-1.1.10/lib/docker/compose/session.rb:266:in `run!': 'up' failed with status 'pid 204 exit 1': .FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/user/Documents/proj/app/new-app/docker-compose.yml' (Docker::Compose::Error)
EXCEPTION: #<Docker::Compose::Error: 'up' failed with status 'pid 204 exit 1': .FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/user/Documents/proj/app/new-app/docker-compose.yml'>
MESSAGE: 'up' failed with status 'pid 204 exit 1': .FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/user/Documents/proj/app/new-app/docker-compose.yml'
```

**Comment by EugenMayer** [comment]
please try https://github.com/EugenMayer/docker-sync-boilerplate in the default folder - does this work. 

Please then remove any extra args from your startup, excludes and really anything and keep it default - does it work?

For me, this looks far more as a docker-compose issue which you did not paste here

**Comment by vteivans** [comment]
OK, I'll try as soon as I can.

But how can it be `docker-compose` problem if it works correctly on its own? When I ran the two commands one after another it works correctly, and this is how I'm developing right now.

**Comment by EugenMayer** [comment]
I think the problem is just your folder layout:
```
 No such file or directory: '/c/Users/user/Documents/proj/app/new-app/docker-compose.yml' (Docker::Compose::Error)
```

I guess your docker-sync.yaml is not located next to the docker-compose.yaml files, but maybe in a subfolder. Could you show us an 

```
/c/Users/user/Documents/proj/app/new-app/
```
?

**Comment by vteivans** [comment]
Nop, they're right next to one another. This is how the `new-app` directory looks like:
```
drwxrwxrwx 1 user user  512 Feb  8 11:15 ./
drwxrwxrwx 1 user user  512 Feb  8 14:08 ../
drwxrwxrwx 1 user user  512 Feb  8 12:51 .docker-sync/
-rwxrwxrwx 1 user user  104 Feb  7 16:33 .dockerignore*
drwxrwxrwx 1 user user  512 Feb  9 15:58 .git/
-rwxrwxrwx 1 user user   58 Feb  8 12:53 .gitignore*
drwxrwxrwx 1 user user  512 Feb  8 10:40 .vscode/
-rwxrwxrwx 1 user user  280 Feb  7 16:33 Dockerfile*
drwxrwxrwx 1 user user  512 Feb  7 17:35 client/
-rwxrwxrwx 1 user user  266 Feb  7 16:33 dev.dockerfile*
-rwxrwxrwx 1 user user  366 Feb  7 17:37 docker-compose-dev.yml*
-rwxrwxrwx 1 user user  329 Feb  7 16:33 docker-compose.yml*
-rwxrwxrwx 1 user user  693 Feb  7 17:36 docker-sync.yml*
-rwxrwxrwx 1 user user 1347 Feb  7 16:33 readme.md*
drwxrwxrwx 1 user user  512 Feb  8 14:03 server/
```

**Comment by EugenMayer** [comment]
The it is finally time to test with the boilerplate 

https://github.com/EugenMayer/docker-sync-boilerplate/tree/master/default

Please do that without modifying any configurations there

Could you also go to your project folder and print `pwd`

I am very curious what `/c/Users/user/Documents/proj/app/new-app/docker-compose.yml` means `/c/` is not common, might be something WSL related, you might have some filesystem / filesystem type or any other overlay issue.

I cannot really guide you any further with Windows WSL - i simply have never used it. Please reassure you have everything right according to the docs

**Comment by vteivans** [comment]
Thanks for such a fast response, I'll try out the boilerplate.

The `/c/` in the path means it's located on `C:` drive in windows. As you may know windows has a table(ish) file system as opposed to *nix. So at the start of the path there is always a drive, where the file is located.

The fact that the path starts with `/c/` is due to new WSL version being able to configure it. Before it was something like `/mnt/c/`. But as far as I understand it's better this way, since all the tooling I've used before always used this type of path - starting with the drive letter.

**Comment by vteivans** [comment]
Tried to run the default boilerplate, same problem with `yml` file not found. Here is the full output:

```bash
user@DESKTOP-90MA50B:/c/Users/User/proj/docker-sync-boilerplate/default$ docker-sync-stack start
/var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/lib/docker-sync/update_check.rb:41: warning: Insecure world writable dir /c/ProgramData/DockerDesktop/version-bin in PATH, mode 040777
       note:  You can also run docker-sync in the background with docker-sync start
     command  docker volume create --opt type=none --opt device="/c/Users/User/proj/docker-sync-boilerplate/default/app" --opt o=bind --name appcode-native-osx-sync
     success  Docker volume for appcode-native-osx-sync created
     success  Watcher disabled by configuration
          ok  Found implicit docker-compose-dev.yml and using it from /c/Users/User/proj/docker-sync-boilerplate/default/docker-compose-dev.yml
          ok  starting compose
.FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/User/proj/docker-sync-boilerplate/default/docker-compose.yml'
#<Thread:0x00007fffdd55dc70@/var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/tasks/stack/stack.thor:50 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        3: from /var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/tasks/stack/stack.thor:51:in `block in start'
        2: from /var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/lib/docker-sync/compose.rb:42:in `run'
        1: from /var/lib/gems/2.5.0/gems/docker-compose-1.1.10/lib/docker/compose/session.rb:94:in `up'
/var/lib/gems/2.5.0/gems/docker-compose-1.1.10/lib/docker/compose/session.rb:266:in `run!': 'up' failed with status 'pid 83 exit 1': .FileNotFoundError:
[Errno 2] No such file or directory: '/c/Users/User/proj/docker-sync-boilerplate/default/docker-compose.yml' (Docker::Compose::Error)
EXCEPTION: #<Docker::Compose::Error: 'up' failed with status 'pid 83 exit 1': .FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/User/proj/docker-sync-boilerplate/default/docker-compose.yml'>
MESSAGE: 'up' failed with status 'pid 83 exit 1': .FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/User/proj/docker-sync-boilerplate/default/docker-compose.yml'


user@DESKTOP-90MA50B:/c/Users/User/proj/docker-sync-boilerplate/default$ docker-sync start
/var/lib/gems/2.5.0/gems/docker-sync-0.5.11.pre.beta2/lib/docker-sync/update_check.rb:41: warning: Insecure world writable dir /c/ProgramData/DockerDesktop/version-bin in PATH, mode 040777
     success  Starting Docker-Sync in the background


user@DESKTOP-90MA50B:/c/Users/User/proj/docker-sync-boilerplate/default$ docker-compose -f docker-compose.yml -f docker-compose-dev.yml up
Creating network "default_default" with the default driver
Pulling app-native-osx (alpine:)...
latest: Pulling from library/alpine
Creating default_app-native-osx_1 ... error

ERROR: for default_app-native-osx_1  Cannot start service app-native-osx: b"error while mounting volume '/var/lib/docker/volumes/appcode-native-osx-sync/_data': error while mounting volume with options: type='none' device='/c/Users/Viesturs/proj/docker-sync-boilerplate/default/app' o='bind': no such file or directory"

ERROR: for app-native-osx  Cannot start service app-native-osx: b"error while mounting volume '/var/lib/docker/volumes/appcode-native-osx-sync/_data': error while mounting volume with options: type='none' device='/c/Users/Viesturs/proj/docker-sync-boilerplate/default/app' o='bind': no such file or directory"
Encountered errors while bringing up the project.
```

**Comment by vteivans** [comment]
I checked the manual again. I followed the points correctly (at least as far as I understand). The only thing to note is that in points `4.` and `5.` I chose to link docker executables from Windows, instead of installing them on WSL Linux distribution (Ubuntu in my case).

**Comment by EugenMayer** [comment]
most probably that is the issue

**Comment by vteivans** [comment]
Yes, that's the issue, I already mentioned it at the begining of this issue. Ths problem is only present when linking windows binaries instead of installing docker in WSL.

**Comment by vteivans** [comment]
I guess I found the issue though - why do we get this error when linking windows binaries. You where right - it's the `docker-compose`. The issue is that, the windows docker compose doesn't understand the `*nix` path, passed to it by `docker-sync-stack`.

I tested this out with two commands:

```bash
user@DESKTOP-90MA50B:/c/Users/User$ docker-compose -f /c/Users/User/proj/docker-sync-boilerplate/default/docker-compose.yml up
.FileNotFoundError: [Errno 2] No such file or directory: '/c/Users/User/proj/docker-sync-boilerplate/default/docker-compose.yml'


user@DESKTOP-90MA50B:/c/Users/User$ docker-compose -f "C:\Users\User\proj\docker-sync-boilerplate\default\docker-compose.yml" up
Every 3s: cat /var/www/index.html                           2019-02-10 14:53:21
app-native-osx_1  |
app-native-osx_1  | cat: can't open '/var/www/index.html': No such file or directory
```

As you can see the first one fails, exactly like the `docker-compose-stack` does but the second one works correctly. This also explains why running the commands separatly works - it's because I'm in the folder with docker files and I don't have to give any path.

Any way, I don't know if it's something you can adjust in the code - translating the path to Windows, when running under WSL. Maybe it's better to simply adjust the manual, to always install the docker on WSL.

**Comment by EugenMayer** [comment]
It would be great if you could put that into the manual!

**Comment by vteivans** [comment]
I could do that, but I'm not sure how to best put it. Originally I thought, maybe just remove the option of symlinking the docker executables from the manual. But maybe an explaining comment would be better.

**Comment by EugenMayer** [comment]
I think both at the same time, explaining is very important so nobody tries again and fails on that

**Comment by Leamsi9** [comment]
I am finding the same issue on a mac, which should be able to understand the *nix path no? Works when decoupled, but with stack it can't find the file. I tried stack because running sync start then compose up no longer seems to be working.  After the initial performance boost I lost the gains and went back to normal speeds after a day or so, with both sync and compose running. I stopped and started both without success.
